packages/date-picker/src/FilterDateRangePicker/utils/getDateRangeDisplayLabel.tsx

```diff
@@ -0,0 +1,20 @@
+import React from "react"
+import { DateRange } from "../../types"
+import { DateRangeDisplayLabel } from "../components/DateRangeDisplayLabel"
+
+export const getDateRangeDisplayLabel = (
```

Comment:
Is this function necessary? Hesitant to have a `getX` function that returns a JSX element

Reply:
I did try to put it in the other one, but TS didn't want to acknowledge the `undefined` return value in the consuming line (`<DateRangeDisplayLabel />` is considered defined), and since the value is passed into `selectedValue` of the button, `null` wasn't a suitable value either as it doesn't make sense api-wise.

If you can think of a better name for this function, I'm happy for that too :)

Reply:
Perhaps the return being `undefined` is the red flag here, could you push this logic upwards and rely on the consuming component to determine whether or not to render the `DateRangeDisplayLabel` component?

Reply:
Hmm... I suppose I could change this into a conditional instead 🤔 But then it would mean I would need more tests on the consuming component too, whereas now I am able to keep it a little more self-contained, which was my preference (although I'm not too strongly tied to this). Thoughts?

Reply:
I think the use case is specific enough that the self-containment doesn't provide much value to the added complexity, and leaving the decision of showing/hiding a label to the parent is a IMO a logical pattern to test for.

But if you had to refactor this using the abstracted pattern, would the conditional get in your way?

Reply:
The abstracted pattern doesn't care about what gets passed into the `selectedValue` - all it cares is that either it exists or not, and if it does, it renders it. 

I did actually try to return `null` in the component and have the button not render if it was `null`, but unfortunately returning `null` in a `JSX.Element` actually still returns a `JSX.Element` just with some attributes set to `null`.

Although now that I say that, I think there is an `isValidElement` function or something which _maybe_ it might be able to help solve it? 🤞🏻

Reply:
Sadly the above did not work :( returning `null` within a JSX.Element is still a valid JSX.Element 😞

Reply:
Moved the conditional to the component :)